x-pack/platform/plugins/private/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -201,16 +196,31 @@ export const reindexServiceFactory = (
 
     let createIndex;
     try {
-      createIndex = await esClient.indices.create({
-        index: newIndexName,
-        settings: {
-          ...settings,
-          // Reindexing optimizations
-          'index.number_of_replicas': 0,
-          'index.refresh_interval': -1,
+      createIndex = await esClient.transport.request<{ acknowledged: boolean }>({
+        method: 'POST',
+        path: `_create_from/${indexName}/${newIndexName}`,
+        body: {
+          // Settings overrides copied from ES datastream reindex logic
+          // https://github.com/elastic/elasticsearch/blob/9c0709f386fee4154e930cb61a02868adebe8572/x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamIndexTransportAction.java#L195-L210
+          settings_override: {
```

Comment:
I know this is a little less than ideal, the current design of this API does not allow another option. Thanks for making the link explicit in code!